### ISSUE TITLE
Cannot initialize Rstudio Container after deleting /tmp 

### ISSUE THREAD

**Issue by Jo-Schie** [rocker-org/rocker#446]
My `/tmp` folder was full of temporary R-folders created from different users and different sessions. I therefore deleted the whole content of `/tmp/` with `rm -rf /tmp/`. 

Turns out this was a very bad move. Now I am not able to login to Rstudio Server anymore. Login Screen appears but then I get an error message. 

![Screenshot 2021-04-07 at 20 36 30](https://user-images.githubusercontent.com/25028884/113917022-fabb9d00-97e0-11eb-8ffa-fd280a2c7102.png)

Different things I tried:

- restarting the docker container
- rebooting the whole VM
- deleting or moving files: 
    - `.local/share/rstudio`
    - `.rstudio`
    - `Rproj.user`
    - `.Rhistory`
    - `.Rdata`

Basically following recommendations from this [Rstudio post](https://support.rstudio.com/hc/en-us/articles/218730228-Resetting-a-user-s-state-on-RStudio-Server) on how to reset a user state and other related blog posts. 

So now I am  running out of ideas. Would be great if someone could give me hint what I could try to make this work again.

**Comment by cboettig** [comment]
can you recreate `/tmp` on the container?  If the container is still up, you should be able to use `docker exec` to enter into it as root and do this.

**Comment by Jo-Schie** [comment]
Yes, I tried that. Unfortunately it did not solve the problem. I was surprised though that /tmp is not automatically recreated as it would be the case in normal Linux I guess. Nevertheless, I tried and it did not work. It's strange in my opinion because `/tmp` should be only for temporary data, so no critical stuff should be inside (that's why I naively deleted it). 

My only explanation is that `Rstudio`somehow saves the state of the sessions and if this information is not available any more than it crashes. Since most of the data is stored in the files I listed above I thought it might solve the problem. The only thing that is stored in `tmp` afaik is the temporary R-folders that look like `Rdfkjfexxlane`or similar and are created at each session startup. 

Is there any way to completely uncouple Rstudio from this "linked information" and just start completly blank sessions? I thought restarting the container or the VM would do this trick but somehow it didn't.

**Comment by eddelbuettel** [comment]
(I actually do not think it _is_ auto-created on Linux. There is other $deity watching over your shoulder and re-creating things.  If you `rm -rf /bin` or any other directory, it is _gone_. For example the various sub-site at stack exchange (unix, sysadmin, ...) are full of questions.  Here are two from em at SO where I tried to help other R users. You need to make `/tmp` also has the right owner, group, permission, and sticky bit.
https://stackoverflow.com/questions/6307836/fatal-error-cannot-mkdir-r-tempdir/6307878#6307878
https://stackoverflow.com/questions/4019113/creating-a-permanent-folder-in-tmp-of-ubuntu-10-04/4019134#4019134 
Good luck! )

**Comment by eddelbuettel** [comment]
(Also, I suggest we transfer this to `versioned2`.  The `rocker` repo has no rstudio container.)

**Comment by cboettig** [comment]
in general when things go pear-shaped I find the best solution is to just delete the container entirely and start fresh.  Obviously back-up any data (can `docker cp` if no other options work) first, though usually I try and volume map my user home dirs so I can discard containers at will.  

a bit orthogonal, but tmp should purge R-related stuff whenever you restart the R session, iirc?
